### ISSUE TITLE
dotnet core support

### ISSUE THREAD

**Issue by jmmk** [haf/Http.fs#125]
Any chance of a dotnet core build? I haven't looked into the transitive dependencies, but I guess all of those would have to support it as well?

**Comment by haf** [comment]
Yes, Hopac supports .Net Core, Expecto does when the branch `netcore` by @adamchester and @cloudroutine is merged (a single admin-esque thing left therein). That's about it, as far as dependencies go.

I don't mind getting pull requests for it.

**Comment by sjalowoi** [comment]
hi ! 

Expecto now support netcore, any chance to have a dotnet core build for Http.fs ? 

Sylvain

**Comment by alex-bogomaz** [comment]
Should it be ```Netstandard 2.0``` port?
```Netstandard 1.6``` is missing a lot of ```HttpWebRequest``` features.

**Comment by haf** [comment]
Sure, 2.0 is fine. I've been holding off, waiting for tooling to stabalise; that's why I haven't gone through all my projects yet. Plus the time-aspect.

**Comment by seanamosw** [comment]
@alex-bogomaz Have you picked this up? If not, I will try an implementation this weekend.

**Comment by alex-bogomaz** [comment]
@seanamosw - probably not. Please feel free to pick up this issue.

**Comment by wklm** [comment]
Hey, when could we expect a release with netcore2 support?

**Comment by haf** [comment]
Albacore v3 https://github.com/Albacore/albacore/wiki was just released in alpha with support for packaging .Net Core projects.

**Comment by seanamos** [comment]
The client itself "works" at the moment on netcore2, other than the HttpWebRequest vs HttpClient issue.

Packaging wise, that also works but I resorted to using paket.template to do it, albacore didn't support it at the time. The actual SDK (`dotnet pack / msbuild /t:Pack`) just doesn't handle building/packaging of projects targeting full .net (net45) & netstandard in the same project across operating systems very well at the moment (immature). It's supposed to "just work" and defer off to mono's msbuild, but there are bugs on both sides.

It should be possible to release, which would unblock .net core 2 projects, with the caveat that until the HttpWebRequest issue is resolved there will be a performance penalty. I'll take a look at that soon if no one else picks it up, I have 2 projects that are blocked as well.

**Comment by whitebear-gh** [comment]
Hi guys, it seems that core support is already added, is it possible to release new nuget package?

**Comment by seanamosw** [comment]
I think the only thing outstanding is updating the README with the API changes.
The HttpWebRequest/HttpClient issue was resolved.

**Comment by haf** [comment]
I'd be happy to give a contributor access to the repo and publishing new nugets. Who wants and have a history of contributions to this repo?

**Comment by whitebear-gh** [comment]
@seanamosw seems to be a perfect candidate :)

**Comment by haf** [comment]
@seanamos I need your nuget username, too =)

**Comment by seanamos** [comment]
@haf I received the invite, is it possible you could send it to this account instead of @seanamosw , that's my work account, I'm just too lazy to swap off it sometimes =) My nuget username is `seanamos`.

**Comment by haf** [comment]
Done!

**Comment by alex-bogomaz** [comment]
Hello, is it possible to release netcore nuget? 

There is a PR for netcore Suave tests, which currently uses full-fw "HTTP.fs". It would be nice to use netcore instead: https://github.com/SuaveIO/suave/pull/671.

**Comment by ivpadim** [comment]
@alex-bogomaz 
@seanamos will be our guy! But I'll be happy to do it... @haf can you give me nuget access my user is ivpadim.

**Comment by haf** [comment]
> Sorry, 'ivpadim' hasn't verified their email account yet and we cannot proceed with the request.

On Nuget.org

**Comment by ivpadim** [comment]
Can you try again? 
Sorry I forgot to verify my email given that I just created the account for this.

**Comment by palutz** [comment]
Not compiling on dotnet core 2.0.2 at the moment. :(

**Comment by palutz** [comment]
Ok.
I have to change this comment.
The lib is actually compiling but still getting the warning about the different fx version:

>  warning NU1701: Package 'Http.fs 4.1.2' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.

**Comment by haf** [comment]
@ivpadim You've been added as the owner now.

**Comment by ivpadim** [comment]
@haf thank you…
Are we ready to deploy? I noticed you removed the gemfile in the build.

**Comment by haf** [comment]
Yes, go ahead if tests pass and travis/appveyor are green :)

**Comment by ivpadim** [comment]
The gemfile is missing and the current version is not compiling... is there a new process to build the package?

**Comment by haf** [comment]
Yes,

https://github.com/haf/Http.fs/blob/master/build.fsx#L101

About compilation; need to know what message you're getting. The stuff I stated working on was not compiling for me.

I have the latest SDK and dotnet cli installed.

**Comment by ivpadim** [comment]
@haf I was able to make it work finally... the v5.0.0 should be up by now...
I had an issue with the Travis CI build somehow the tests execution port is not getting released (fixed by only executing the dotnetcore version) and the appveyor build is not working at all.

I will be updating the doc this week.

**Comment by ivpadim** [comment]
The dotnet core version was released a few days ago. If you encounter any issue please open a new issue.
